app/models/region.rb

```diff
@@ -65,14 +66,16 @@ def protected_areas_per_iucn_category
       end
     end
 
-    processed_data = region_data.map{ |key,value| {
-      "iucn_category_name" => key,
-      "count" => value["count"],
-      "percentage" => 100 * value["count"] / total_region_count.reduce(0, :+)
+    processed_data = correct_order.map do |key,value|
+      next if region_data[key].nil?
+      {
+        "iucn_category_name" => key,
+        "count" => region_data[key]["count"],
+        "percentage" => 100 * region_data[key]["count"] / total_region_count.reduce(0, :+)
       }
-    }
+    end
 
-    processed_data
+    processed_data.compact
```

Comment:
Do we need `.compact` here? Ideally we are skipping all `nil` values in the previous loop

Reply:
True @Levia! I don't know why but I got nil values coming through, the loop must need changing to remove this `compact`.

Reply:
It appears the `next if` is outputting a nil value when that occurs.